File: C++/Manual 08/Question01.cpp
#include <iostream>
using namespace std;

// Function to swap two integers
void swap(int &x, int &y)
{
    int temp = x;
    x = y;
    y = temp;
}

// Function to perform selection sort
void Sort(int a[], int n)
{
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++)
    {
        min_idx = i;
        for (j = i + 1; j < n; j++)
        {
            if (a[j] < a[min_idx])
                min_idx = j;
        }
        if (min_idx != i)
        {
            swap(a[min_idx], a[i]);
        }
    }
}

// Function to print the array
void printArray(int a[], int size)
{
    int i;
    for (i = 0; i < size; i++)
        cout << a[i] << " ";
    cout << endl;
}

int main()
{
    int a[100], n;
    cout << "Enter Number of Elements: ";
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        cout << "Enter " << i + 1 << "th Term: ";
        cin >> a[i];
    }
    Sort(a, n);
    cout << "Sorted Array: \n";
    printArray(a, n);
    return 0;
}

File: Python/Manual 08/Question01.py
import random

myArray = []

for i in range(10):
    myArray.append(random.randint(0, 100))


def swap(i, j):
    return j, i


def selectionSort(array):
    n = len(array)
    for i in range(n):
        min_idx = i
        for j in range(i + 1, n):
            if array[j] < array[min_idx]:
                min_idx = j

        if min_idx != i:
            array[i], array[min_idx] = swap(array[i], array[min_idx])


selectionSort(myArray)

print("Sorted array:", myArray)
